modules/internal/ChapelBase.chpl

```diff
@@ -2150,23 +2150,45 @@ module ChapelBase {
   }
 
   // casting to unmanaged?, no class downcast
+  pragma "last resort"
+  @deprecated("casting from a managed class to an 'unmanaged?' is deprecated")
```

Comment:
"borrowed" is not a "managed" category, if I understand correctly. At the very least the deprecation text should be different.

Reply:
I am also unhappy with the deprecation message. But due to how the compiler resolves casts (compiler/resolution/functionResolution.cpp:3087), I could not find a way to get the true type being casted from, it always returned `borrowed`. I decided that the efforts to make it work were not worth it for a 6 month deprecation. I am definitely open to better wordsmithing here though